src/types/library/library_item.rs

```diff
@@ -46,6 +46,14 @@ impl LibraryItem {
     pub fn is_in_continue_watching(&self) -> bool {
         self.r#type != "other" && (!self.removed || self.temp) && self.state.time_offset > 0
     }
+    #[inline]
+    pub fn progress(&self) -> f64 {
```

Comment:
- [ ] Replace this progress calculation in `core-web`:
https://github.com/Stremio/stremio-core-web/blob/development/src/model/serialize_library.rs#L110-L114